### PR TITLE
External API Consumption for Zipcode to Location

### LINKED ISSUE #66: Service: Make external api consumption of zip code to location

### PR DESCRIPTION
**Pull request by priyapower**
# This PR:
---

## Changes
- Adds a service file for external api consumptions that deal with the users zip code
- Functions:
  - `zip_to_location`: converts zip code to a city and state
    - Future Case: we may need just the state or city delivered separate
  - `zip_to_avg_home`: converst zip code to a median home price by location
    - **NOT FINISHED** - this needs authorization from api sites (currently attempting to use HouseCanary)
- [X] Adds code for functionality
- [ ] Adds a test
- [ ] Updates Documentation/ReadMe
- [X] Adds dependencies
  - `dotenv`
  - `requests`
- [ ] Refactors code
- [ ] Fixes a bug

## Checklist
**Please check off what you can for this PR:**
- [ ] This code passes all tests on local machine
- [X] This code is DRY
- [X] This code follows OOP
- [X] This code, if updates to functionality were made, updates all corresponding files

**The "I have" section is a double check for PR specific information:**
- [X] I have tagged reviewers for this PR
- [X] I have checked what branch I am merging into (current expectation is `it1_development`)
- [X] I have added this PR to the project
- [X] I have added this PR to the correct milestone
- [ ] I have updated documenation if changes were made that affect setup/running

## Linked tickest / issues 
Closes #66

### CODE REVIEW

**Review by timkeresey — APPROVED** [review]
This is what will give us functionality to take in the user's zip and give their location in the report, right?

File: Pipfile [hunk 1]
```diff
@@ -9,6 +9,8 @@ flask-mongoengine = "*"
 "pry.py" = "*"
 gunicorn = "*"
 flask-cors = "*"
+requests = "*"
+python-dotenv = "*"
```
Comment (Arique1104) [inline Pipfile:13]: Glad these were officially added to our project's pipfile!

**Comment by priyapower** [comment]
@timkeresey - Yes! This is the functionality for zip to location! We are still attempting to figure out zip to avg home price but we are still struggling to find an external api that will give us a key

File: api/models/report.py [hunk 2]
```diff
@@ -45,11 +46,10 @@ def mortgage_rate(self):
         return round(rate, 4)
 
     def city_state(self):
-        # 80209 entered for zip
-        return "Denver, CO"
+        return zip_to_location(self.zipcode)
```
Comment (Arique1104) [inline api/models/report.py:49]: This looks so clean and simple!

File: api/models/report.py [hunk 3]
```diff
@@ -45,11 +46,10 @@ def mortgage_rate(self):
...
 
     def home_price_by_zip(self):
-        return "$834,278"
+        return zip_to_avg_home(self.zipcode)
```
Comment (Arique1104) [inline api/models/report.py:52]: Can't wait for this to be "on!"  I hope we find something to populate it!

File: api/services/zip.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    url = f'http://www.zipcodeapi.com/rest/{api_key}/info.{format}/{zip}/{units}'
+
+    response = requests.get(url)
+    json_response = json.loads(response.text)
+    city = json_response['city']
+    state = json_response['state']
+    return f'{city}, {state}'
+
+def zip_to_avg_home(zip):
+    return 350000
```
Comment (Arique1104) [inline api/services/zip.py:19]: This is hardcoded for now until we determine what our plan is to gather this information.

File: app.py [hunk 5]
```diff
@@ -14,17 +14,20 @@
...
-    'host': os.environ['HOST'],
-    'username': os.environ['USERNAME'],
-    'password': os.environ['PASSWORD']
-}
+# app.config['MONGODB_SETTINGS'] = {
+#     'db': os.environ['DBNAME'],
+#     'host': os.environ['HOST'],
+#     'username': os.environ['USERNAME'],
+#     'password': os.environ['PASSWORD']
+# }
```
Comment (Arique1104) [inline app.py:23]: We need to turn this back on once we merge this to main.

**Review by Arique1104 — APPROVED** [review]
Looks great!  I'm merging it!

**Pull request merged**